### DISCUSSION TITLE
Search Widget standardization

### DISCUSSION THREAD

**Discussion by pajlada** [Chatterino/chatterino2#3428] [Ideas]
We're starting to get a few more search inputs around Chatterino, it would be good to standardize their behaviour through a `SearchInputWidget` of some kind

1) When the Search Input is selected and is not empty, pressing Escape should be intercepted and clear the search entry
2) The clear button behaviour of the input should be enabled (See https://github.com/Chatterino/chatterino2/pull/3403)
3) The SearchInputWidget must force the developer to specify a placeholder text

CC @zneix

**Comment by Mm2PL** [comment]
It would also be nice if it forced the developer to use the hotkeys system and add a `search` action.
